runtimes/java-18.0/src/main/java/io/openruntimes/java/Server.java

```diff
@@ -9,53 +9,158 @@
+
+        if(hostHeader.contains(":")) {
+            host = hostHeader.split(":")[0];
+            port = Integer.parseInt(hostHeader.split(":")[1]);
+        } else {
+            host = hostHeader;
+            port = Integer.parseInt(defaultPort);
+        }
+
+        String path = req.path();
+        String queryString = req.query();
+        Map<String, String> query = new HashMap<String, String>();
+
+        for (String param : queryString.split("&")) {
+            String[] pair = param.split("=", 2);
+            
+            if(pair.length >= 1 && pair[0] != null && !pair[0].isEmpty()) {
+                String value = pair.length == 2 ? pair[1] : "";
+                query.put(pair[0], value);
+            }
         }
-        if (!req.headers().get("x-internal-challenge").equals(System.getenv("INTERNAL_RUNTIME_KEY"))) {
-            return resp.code(500).result("Unauthorized");
+
+        String url = scheme + "://" + host;
+
+        if(port != Integer.parseInt(defaultPort)) {
+            url += ":" + String.valueOf(port);
         }
 
-        Wrapper codeWrapper = new Wrapper();
-        RuntimeRequest request = new RuntimeRequest(req);
-        RuntimeResponse response = new RuntimeResponse();
+        url += path;
+
+        if(!queryString.isEmpty()) {
+            url += "?" + queryString;
+        }
+
+        RuntimeRequest runtimeRequest = new RuntimeRequest(bodyString, body, headers, method, url, scheme, host, path, port, queryString, query);
+        RuntimeResponse runtimeResponse = new RuntimeResponse();
+        RuntimeContext context = new RuntimeContext(runtimeRequest, runtimeResponse);
 
-        ByteArrayOutputStream outStream = new ByteArrayOutputStream();
-        ByteArrayOutputStream errStream = new ByteArrayOutputStream();
-        PrintStream userOut = new PrintStream(outStream);
-        PrintStream userErr = new PrintStream(errStream);
         PrintStream systemOut = System.out;
         PrintStream systemErr = System.err;
+        
+        ByteArrayOutputStream customstdStream = new ByteArrayOutputStream();
+        PrintStream customstd = new PrintStream(customstdStream);
+        System.setOut(customstd);
+        System.setErr(customstd);
 
-        System.setOut(userOut);
-        System.setErr(userErr);
+        RuntimeOutput output = null;
 
         try {
-            RuntimeResponse userResponse = codeWrapper.main(request, response);
-            Map<String, Object> output = new HashMap<String, Object>();
-            output.put("response", userResponse.data);
-            output.put("stdout", outStream.toString());
-            return resp.result(RuntimeResponse.gson.toJson(output));
+            Class classToLoad = Class.forName("io.openruntimes.java.Tests");
+            Method classMethod = classToLoad.getDeclaredMethod("main", RuntimeContext.class);
+            Object instance = classToLoad.newInstance();
+            output = (RuntimeOutput) classMethod.invoke(instance, context);
```

Comment:
Does this need to be updated from the `Tests` class? You shouldn't need to use reflection here

Reply:
Good catch, I hardcoded this and forgot about it. Updated.